### PR TITLE
Fix error message because of macro

### PR DESCRIPTION
**Pull request by kdheepak**
Related to #125

### CODE REVIEW

**Comment by kdheepak** [comment]
Tests pass for me locally.

Screen Shot 2021-05-08 at 11 16 57 AM

However I don't know if this error case is being tested. Does anyone know how to make one of the `Lib.XPRS...` functions fail consistently so that we can add it as a test?

**Comment by kdheepak** [comment]
I've added a test that always fails. It would be nice to test that the error message string contains `91 Error: No problem has been input` but I think this is fine for now.

```
    prob = Xpress.XpressProblem()
    @test_throws Xpress.XpressError Xpress.copyprob(prob, prob)
```

File: src/api.jl [hunk 1]
```diff
@@ -73,7 +73,7 @@ Initializes the Optimizer library. This must be called before any other library
 
 """
 function init()
-    @checked Lib.XPRSinit(C_NULL)
+    Lib.XPRSinit(C_NULL)
```
Comment (joaquimg) [inline src/api.jl:76]: We could documente in the macro def why some functions wont use it.

**Comment by rafabench** [comment]
> I've added a test that always fails.
> 
> ```
>     prob = Xpress.XpressProblem()
>     @test_throws Xpress.XpressError Xpress.copyprob(prob, prob)
> ```
> 
> It would be nice to test that the error message string contains `91 Error: No problem has been input` but I think this is fine for now. If anyone knows how to do that I can update the test.

I think this is a way to test this Xpress error.

```
msg = "Unable to call `Xpress.copyprob`:\n\n91 Error: No problem has been input.\n"
@test_throws Xpress.XpressError(32,msg) Xpress.copyprob(prob, prob)
````

File: src/api.jl (hunk 1 above)
Comment (kdheepak) [reply src/api.jl:76]: Resolved in https://github.com/jump-dev/Xpress.jl/pull/126/commits/f7fc8fb66bf26225aa680d6e65a2a06dbad89a69.

**Comment by kdheepak** [comment]
Thanks @rafabench I didn't know about that syntax! I guess I should have read the documentation 😅. 

I've updated the PR.

File: src/api.jl [hunk 2]
```diff
@@ -2787,7 +2787,9 @@ Returns the error message corresponding to the last error encountered by a libra
 
 """
 function getlasterror(prob::XpressProblem)
-    @invoke Lib.XPRSgetlasterror(prob, _)::String
+    out = Cstring(pointer(Array{Cchar}(undef, 512)))
+    s = Lib.XPRSgetlasterror(prob, out)
+    s == 0 ? unsafe_string(out) : "Unable to get last error"
 end
```
Comment (odow) [inline src/api.jl:2793]: This should probably be 
```Julia
function getlasterror(prob::XpressProblem)
    @invoke Lib.XPRSgetlasterror(prob, _)::String
    buffer = Array{Cchar}(undef, 512)
    buffer_p = pointer(buffer)
    GC.@preserve buffer begin
        s = Lib.XPRSgetlasterror(prob, buffer_p)
        return s == 0 ? unsafe_string(buffer_p) : "Unable to get last error"
    end
 end
```

Comment (kdheepak) [reply src/api.jl]: I think you are right! I've made this change along with another potentially similar issue in https://github.com/jump-dev/Xpress.jl/pull/126/commits/8833da4ddfbb87ebade64a71fcf1939ba9a04a97.

Comment (kdheepak) [reply src/api.jl]: Added more changes in here: https://github.com/jump-dev/Xpress.jl/pull/126/commits/d3fe3421dc420bfea015e389dbd03cb34cf05de9

**Review by joaquimg — APPROVED** [review]

**Comment by joaquimg** [comment]
Good for me!

**Pull request merged**